handlers/sf-emails-to-s3-exporter/src/main/scala/com/gu/sf_emails_to_s3_exporter/Handler.scala

```
@@ -13,26 +13,31 @@ object Handler extends LazyLogging {
   }
 
   def handleRequest(): Unit = {
-    (for {
+
+    val emailsFromSF = for {
       config <- SalesforceConfig.fromEnvironment.toRight("Missing config value")
-    } yield for {
-      sfAuthDetails <- decode[SfAuthDetails](auth(config).getOrElse(throw new RuntimeException("Error in SF Authentication")))
+      authentication <- auth(config)
+      sfAuthDetails <- decode[SfAuthDetails](authentication)
       emailsFromSF <- getEmailsFromSfByQuery(sfAuthDetails)
-    } yield processEmails(sfAuthDetails, emailsFromSF)) match {
-      case Left(abc) => { println(abc) }
-      case Right(fff) => { println(fff) }
+    } yield processEmails(sfAuthDetails, emailsFromSF)
+
+    emailsFromSF match {
+      case Left(ex) => {
+        logger.error("Error: " + ex)
```

After logging the exception, you probably want to throw a runtime exception to signal to the lambda environment that the lambda has failed.   Otherwise an alarm based on failure won't be triggered.
```
throw new RuntimeException(ex)
```
This is probably the only situation in which you'd ever need to throw an exception.

---

Done